Create a Contributor Guide in docs

willingc opened the pull request:
This is an initial step of moving information contained in CONTRIBUTING.md to a Contributor Guide to be hosted as part of the nbgrader docs hosted on Read the Docs.

The file structure and indices have been set up. The docs still need conversion from markdown to rst.

As discussed in #401 and #392.

jhamrick commented:
Great!

You can convert the markdown to rst with pandoc pretty easily, I think `pandoc --from markdown --to rst CONTRIBUTING.md`.

willingc commented:
@jhamrick Ready for review. I converted the markdown to rst, added subsection titles, and made some edits for organizational flow.

On a separate topic, I've been building with Sphinx and the make file. I forgot there was invoke docs. I did get an error about invoke docs running and I'm wondering if you noticed that too. Since this pr has no ipynb files, there was no need to invoke clear_docs. I will investigate when I return.

jhamrick commented on docs/source/contributor_guide/installation_developer.rst:
```diff
@@ -0,0 +1,53 @@
+You will use ``pip3`` to install `flit <https://github.com/takluyver/flit>`_
+and then use ``flit`` to install and build nbgrader::
+
+    pip3 install flit
```
Thanks to #409 this can just be one line instead of two, and it can be:

```
python setup.py develop
```

jhamrick replied:
You can also remove the bit about python 3, see the changes here: https://github.com/jupyter/nbgrader/pull/409/files#diff-6a3371457528722a734f3c51d9238c13L36

jhamrick commented on docs/source/contributor_guide/documentation.rst:
````diff
@@ -0,0 +1,77 @@
+   documentation notebooks will be converted to rst, and example notebooks will
+   be converted to html) using the ``build_docs.py`` script.
+2. The command line documentation is automatically generated using the
+   ```autogen_command_line.py`` and ``autogen_config.py`` scripts.
````
Extra backtick here

willingc replied:
Good catch

jhamrick commented:
This looks great, thanks! Just a few small comments that I left inline.

willingc commented:
@jhamrick Changes made. I like the simplified install :dog:

willingc commented:
@jhamrick Sorry about the backticks. Your tip on markdown vs. rst was appreciated. Honestly, the oversight was me rushing on the :airplane: . Hopefully, channeling the :turtle: for a bit instead of the :rabbit2: has resulted in better attention to detail ;)

jhamrick commented:
Looks good, thanks!
